### PR TITLE
Ui improvement

### PR DESCRIPTION
**Pull request by empratyush**
- Fix "uninstall" menu item for system apps.
- Fix "open button" navigating to details screen instead of opening app in some cases.

### CODE REVIEW

File: app/src/main/java/org/grapheneos/apps/client/utils/network/MetaDataHelper.kt [hunk 1]
```diff
@@ -224,6 +227,21 @@ class MetaDataHelper constructor(context: Context) {
         return result
     }
 
+    private fun isSystemApp(packageName: String, ogPackageName: String? = null): Boolean {
+
+        val sigFlags = PackageManager.GET_SIGNING_CERTIFICATES
```
Comment (muhomorr) [inline app/src/main/java/org/grapheneos/apps/client/utils/network/MetaDataHelper.kt:232]: Why obtain the signing certificates? They aren't used in this method.

File: app/src/main/java/org/grapheneos/apps/client/utils/network/MetaDataHelper.kt [hunk 2]
```diff
@@ -224,6 +227,21 @@ class MetaDataHelper constructor(context: Context) {
...
+        return try {
+            val pmInfo = context.packageManager.getPackageInfo(packageName, sigFlags)
+            (pmInfo.applicationInfo.flags and ApplicationInfo.FLAG_SYSTEM) != 0
+        } catch (e: PackageManager.NameNotFoundException) {
+            if (ogPackageName != null && ogPackageName.isNotEmpty()) {
+                return isSystemApp(packageName)
```
Comment (muhomorr) [inline app/src/main/java/org/grapheneos/apps/client/utils/network/MetaDataHelper.kt:238]: Should be `isSystemApp(ogPackageName)`

**Review by muhomorr — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/grapheneos/apps/client/utils/network/MetaDataHelper.kt (hunk 1 above)
Comment (empratyush) [reply app/src/main/java/org/grapheneos/apps/client/utils/network/MetaDataHelper.kt]: fixed.

**Comment by muhomorr** [comment]
There's a duplicate `isSystemApp()` method in App.kt:
https://github.com/GrapheneOS/Apps/blob/main/app/src/main/java/org/grapheneos/apps/client/App.kt#L365

Would be good to remove it by making the new `isSystemApp()` method accept `Context` as a parameter and moving the new method out of `MetadataHelper` class

**Comment by empratyush** [comment]
done.

File: app/src/main/java/org/grapheneos/apps/client/utils/network/MetaDataHelper.kt [hunk 3]
```diff
@@ -224,6 +227,19 @@ class MetaDataHelper constructor(context: Context) {
         return result
     }
 
+    private fun isSystemApp(packageName: String, ogPackageName: String? = null): Boolean {
+        return try {
+            val pmInfo = context.packageManager.getPackageInfo(packageName, 0)
+            (pmInfo.applicationInfo.flags and ApplicationInfo.FLAG_SYSTEM) != 0
+        } catch (e: PackageManager.NameNotFoundException) {
+            if (ogPackageName != null && ogPackageName.isNotEmpty()) {
```
[the remark this answers is not in the document]
Comment (empratyush) [reply app/src/main/java/org/grapheneos/apps/client/utils/network/MetaDataHelper.kt]: metadata can have empty/blank original package name, check for isNotEmpty is fine IMO.

**Comment by thestinger** [comment]
Going to be incorporated as part of https://github.com/GrapheneOS/Apps/pull/151 instead due the conflicts.

**Pull request merged**